Rock.Rest/Controllers/BinaryFilesController.Partial.cs

```diff
@@ -0,0 +1,166 @@
+// limitations under the License.
+// </copyright>
+//
+using System;
+using System.IO;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Web;
+using System.Web.Http;
+using Rock.Data;
+using Rock.Model;
+using Rock.Rest.Filters;
+using Rock.Utility;
+
+namespace Rock.Rest.Controllers
+{
+    /// <summary>
+    ///
+    /// </summary>
+    public partial class BinaryFilesController
+    {
+        /// <summary>
+        /// Uploads a file and stores it as a binary file
+        /// </summary>
+        /// <param name="binaryFileTypeId"></param>
+        /// <returns></returns>
+        [Authenticate, Secured]
+        [HttpPost]
+        [System.Web.Http.Route( "api/BinaryFiles/Upload" )]
+        public HttpResponseMessage Upload( Guid binaryFileTypeGuid )
+        {
+            try
+            {
+                var rockContext = new RockContext();
+                var binaryFileType = new BinaryFileTypeService( rockContext ).Get( binaryFileTypeGuid );
+
+                if ( binaryFileType == null )
+                {
+                    GenerateResponse( HttpStatusCode.BadRequest, "Invalid binary file type guid" );
+                }
+
+                return Upload( binaryFileType.Id );
+            }
+            catch ( HttpResponseException exception )
+            {
+                return exception.Response;
+            }
+            catch
+            {
+                return new HttpResponseMessage( HttpStatusCode.InternalServerError )
+                {
+                    Content = new StringContent( "Unhandled exception" )
+                };
+            }
+        }
+
+        /// <summary>
+        /// Uploads a file and stores it as a binary file
+        /// </summary>
+        /// <param name="binaryFileTypeId"></param>
+        /// <returns></returns>
+        [Authenticate, Secured]
+        [HttpPost]
+        [System.Web.Http.Route( "api/BinaryFiles/Upload" )]
+        public HttpResponseMessage Upload( int binaryFileTypeId )
+        {
+            try
+            {
+                var rockContext = new RockContext();
+                var context = HttpContext.Current;
+                var files = context.Request.Files;
+                var uploadedFile = files.AllKeys.Select( fk => files[fk] ).FirstOrDefault();
+                var binaryFileType = new BinaryFileTypeService( rockContext ).Get( binaryFileTypeId );
+
+                if ( uploadedFile == null )
+                {
+                    GenerateResponse( HttpStatusCode.BadRequest, "No file was sent" );
+                }
+
+                if ( binaryFileType == null )
+                {
+                    GenerateResponse( HttpStatusCode.InternalServerError, "There is no person image file type" );
+                }
+
+                var currentUser = UserLoginService.GetCurrentUser();
+                var currentPerson = currentUser != null ? currentUser.Person : null;
+
```

Comment:
The base ApiController class has a GetPerson() method that should be used to get the person.